lib/browser/objects-registry.js

```diff
@@ -20,6 +20,7 @@ class ObjectsRegistry {
   add(webContents, obj) {
     // Get or assign an ID to the object.
     let id = this.saveToStorage(obj);
+    this.storage[id].count++;
```

Comment:
Ah, I think I see where our understanding differs.
Let's look at the internal electron use for a minute: https://github.com/atom/electron/blob/master/lib/renderer/override.js#L10-L13
Now, __even if there is another user of `remote.getCurrentWindow()`__, `OnObjectGC` will still be called for the `currentWindow` reference to `browserWindow`.

I've got a theory:
`setDestructor()` applies to the actual _javascript_ variable: `currentWindow`. In the example in this comment, `OnObjectGC` will be called when `currentWindow` goes out of scope (which it will, it's just alive for like 4 lines). The key part: even if another part of the program calls `remote.getCurrentWindow()`, that will produce a __different__ javascript variable, with its own scope and lifetime. `setDestructor()` only applies to one javascript variable at a time.

So, `setDestructor()` is working properly, but it doesn't care about the amount of electron-style references to `browserWIndow`. It only cares that a javascript variable (`currentWindow) went out of scope and was GC'd